app.rb

```diff
@@ -75,12 +75,12 @@ def auth_server_uri
   end
 
   post '/consume/?' do
-    response = OneLogin::RubySaml::Response.new(params[:SAMLResponse])
+    response = OneLogin::RubySaml::Response.new(
+      params.fetch(:SAMLResponse), settings: saml_settings
```

Comment:
This should be `'SAMLResponse'`, as a String, not a Symbol. I just tested it locally.

Reply:
The only thing that I can think of that would explain how this ever worked is that the assertion was not encrypted at some point.  The only scenario where the `ruby-saml` gem expects the settings to be available during initialization is if the assertion is encrypted. If the assertion is not encrypted, you can set the settings after the fact and then call `is_valid?`.

Reply:
Thanks, good catch.